src/test_report.py

```diff
@@ -96,32 +95,48 @@ def send_mail(self, email_msg, email_server):
         else:
             self._logger.log_message(logging.INFO, "Email Sent.")
 
+    def get_test_analysis(self, nodes):
+        total_runs = len(nodes)
+        total_failures = sum(node['status'] == "fail" for node in nodes)
+        total_timeouts = sum(node['status'] == "timeout" for node in nodes)
+        return total_runs, total_failures, total_timeouts
+
+    def create_test_report(self, root_node, child_nodes):
+        total_runs, total_failures, total_timeouts = \
+                    self.get_test_analysis(child_nodes)
+
+        template_env = jinja2.Environment(
+                            loader=jinja2.FileSystemLoader("./config/reports/")
+                        )
+        template = template_env.get_template("test-report.jinja2")
+        subject_str = (f"{root_node['revision']['tree']}/"
+                       f"{root_node['revision']['branch']} "
+                       f"{root_node['revision']['describe']}: {total_runs}"
+                       f"runs {total_failures} fails ({root_node['name']})")
```

Comment:
```suggestion
        subject_str = (f"{root_node['revision']['tree']}/\
   {root_node['revision']['branch']} \
   {root_node['revision']['describe']}: \
   {total_runs} runs {total_failures} fails \
   ({root_node['name']})")
```

Reply:
This is not resolved yet.
Did you test it on your local?
We are still missing whitespace at each line here used for formatting.

```suggestion
{root_node['revision']['branch']} \
{root_node['revision']['describe']}: \
{total_runs} runs {total_failures} fails \
```

Reply:
@JenySadadia  Yes, I did test! It was working, but you're right I miss the whitespaces and now it is updated.